src/adjtrans.jl

```diff
@@ -92,7 +92,7 @@ function *(op :: AdjointLinearOperator{T,S}, v :: AbstractVector{U}) where {T,S,
   ishermitian(p) && return p * v
   if p.ctprod !== nothing
     increase_nctprod(p)
-    return p.ctprod(v)::Vector{promote_type(T,U)}
+    return p.ctprod(v)::typeof(v).name.wrapper{promote_type(T,U), typeof(v).parameters[2]}
```

Comment:
`typeof(v).parameters[2]` represents the dimension of the Array if it's dense or the type of Integer that is used for storing coefficients when it's sparse.